pkg/bot/discord.go

```diff
@@ -345,6 +363,19 @@ func (b *Discord) formatMessage(msg interactive.CoreMessage) (*discordgo.Message
 	}, nil
 }
 
+func (b *Discord) shutdown() {
+	b.shutdownOnce.Do(func() {
+		b.log.Info("Shutting down discord message processor...")
+		close(b.messages)
+		b.discordMessageWorkers.Wait()
+
+		err := b.api.Close()
```

Comment:
first we need to close the connection to make sure that there won't be any new messages coming, otherwise you can write to already closed `b.messages` causing panic.